### PR TITLE
Private Events

### PR DESCRIPTION
**Pull request by sumon0002001**
In this project, we have built an application named Private Events and that behaves similar to Eventbrite.
The main goal was to put in practice the main concepts of Associations in rails.

### CODE REVIEW

**Comment by descholar-ceo** [comment]
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

File: app/views/events/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+<% if !Event.any? %>
+  <h2>No events</h2>
+<% else %>
+  <div class="upcoming-cont">
+    <% if @upcoming.any? %>
+      <h2>Upcoming Events</h2>
+      <ul class = "upcoming">
+        <% @upcoming.each do |e| %>
+            <li>
+              <% if session[:current_user] %>
```
Comment (descholar-ceo) [inline app/views/events/index.html.erb]: There should no any logic in views

File: app/views/events/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+<ul>
+  <% @event.attendee.each do |a| %>
+    <li>
+      <%= a.username %>
+    </li>
+  <% end %>
+</ul>
+<% end %>
+
+<% if @event.date.localtime.strftime("%m/%d/%Y %H:%M") > @date %>
```
Comment (descholar-ceo) [inline app/views/events/show.html.erb]: Avoid logics in views

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @sumon0002001  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice and it fulfill almost all requirements, congratulations on the good work you have done so far.
:+1: I liked that you the extra efforts you put in auth system and great associations, keep it up :muscle:.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [ ] Event's index page displays past and upcoming events lists ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [ ] The user's show page displays events the user has created and events that the user is attending to separated in past and upcoming ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from Odin)
- [ ] Files not related to the project should not be committed and added to .gitignore (e.g. .DS_Store) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/auth/sign-in/?next=/library/fast-track-curriculum/69047/path/step/66243643/))
- [ ] You have a lot of logics in views: There shoud be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from `Pathwright`
Resources
    - [ ] View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb) => To help you unerstand what to change: Make sure that there are no if conditions, or loops in views

Resources that can help you: 
:point_right: [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code/)
:point_right: [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)

___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 

Great job so far @sumon0002001 :clap: :clap: :clap: :+1: :100: , though you still have some changes :hammer: :hammer_and_wrench: :toolbox: to take care of in order to get your project approved :checkered_flag:.

Please check the comments below for more details:
:point_down: :point_down: :point_down: :point_down: :point_down: 

## Required Changes :red_circle::

- [x] You have several errors on your Event#index page check it and fix them, please always run your project before you request for a code review:
image

- [x] Please remove all logic from your views as requested by the previous TSE.

### Please make all the request changes before you submit another code review.
## Happy Coding :tophat: 
_Feel free to contact me in Slack if you have any questions or need help_
## _Zakariae El Mejdki_

**Review by Georjane — CHANGES_REQUESTED** [review]
## Status: Changes Required 🛑 🔧 
### Hi there Team ✋🏽 
Very nice work you have done there. Your project meets all the basic functionality of the Private Events.

- [ ] However, your views contain a lot of logic in them. By logic, I mean the so many if and else statements in your code. These conditionals can be turned to methods and placed in the Helper folders where you can easily call them in your logic. Read [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) article for more information on where to place your code.

_Feel free to reach out to me on Slack if you have any questions, Witah Georjane_ 🤝 
Good luck on your project and Happy Coding 💻 

**Witah Georjane
Code Reviewer at Microverse Inc
[portfolio](https://witahgeorjane.tech/) | [linkedIn](https://www.linkedin.com/in/witah-georjane/)**

**Review by oloomoses — APPROVED** [review]
# Approved: Congratulations :tada: :100: :muscle: 
### You have done well on Implementing the required changes. Your project meets all the requirements.
### You can merge this pull request and, I wish you the best on your next project:rocket:

**Pull request merged**